src/collections/blog/2023/2023-05-31-interview-experience/index.mdx

```diff
@@ -0,0 +1,77 @@
+---
+title: "Charting Uncharted Waters: My Transformative LFX Internship at Meshery"
+subtitle: "Navigating Challenges, Discovering Success, and Redefining Boundaries"
+date: 2023-05-31 05:30:05 -0530
+author: Azanul Haque
+thumbnail: ./azan-blog.png
+darkthumbnail: ./azan-blog.png
+category: Community
+description: "Stepping Outside my Comfort Zone: My LFX Internship Adventure in Go and Kubernetes at Meshery."
+tags:
+  - Open Source
+  - Experience
+  - Layer5
+type: Blog
+resource: false
+published: true
+---
+
+import { BlogWrapper } from "../../Blog.style.js";
+
+
+<BlogWrapper>
+
+## I. Introduction
+
+   In a world where technology is evolving at an unprecedented pace, the significance of exploring new technologies cannot be overstated. As aspiring professionals, we find ourselves standing on the precipice of innovation, eager to delve into uncharted territories and seize the transformative opportunities that lie ahead. Nothing accelerates learning in a field more than gaining practical, hands-on experience. 
+   The purpose of the blog post is to provide the reader a sneak peek into a LFX internship and what transitioning to a completely new tech stack is like.
+   Successful completion of my tasks at Meshery during this internship required me to have deep knowledge of Go, Docker, Kubernetes, Helm and a pinch of ReactJs. 3 months ago, I couldn't say I had any of these. This is me sharing my journey of learning through hands-on experiences and adapting on the fly.
+
+## II. Prelude to the Internship
+
+   I wanted to participate in programs like GSoC and LFX Mentorship ever since I got to know about them. These programs focused on helping students  and early-career professionals with valuable opportunities to gain practical experience, enhance their skills, and contribute to open-source projects. These programs aim to bridge the gap between academic learning and real-world application by offering mentorship and guidance from experienced professionals in the industry.
```

Comment:
```suggestion
   I have wanted to participate in programs like GSoC and LFX Mentorship ever since I got to know about them. These programs focused on helping students  and early-career professionals with valuable opportunities to gain practical experience, enhance their skills, and contribute to open-source projects. These programs aim to bridge the gap between academic learning and real-world application by offering mentorship and guidance from experienced professionals in the industry.
```

Reply:
I already competed LFX, why would I use past continuous tense.

Reply:
Past continuous tense is often used while telling a story to convey that something continued for over a period of time, not a specific moment in time. Did you have this desire once and only once at a specific moment in time or was this a general feeling of desire that built as you learned more about the programs and the valuable experiences that you make gain as you force reviewers to teach you grammar instead of taking your mentor's word for it? My sense was that you'd had this feeling more than once (e.g. "This time was a bit different...").

Reply:
> taking your mentor's word for it?

Ok boss! Sky sure is green and horses do fly.

Commenting this here just for some poor soul still open to seeing error in their ways.
Past simple: I wanted (started in past, ended in past).
Past continuous 🤣 (used when a specific time is referenced): I was wanting (started in past, continued in past, ended in past).
Present perfect: I have wanted (started in the past, continues to present).

A simple exercise to figure this out would be to reference the pharaohs.
Pharaohs ruled Egypt. ✅ 
Pharaohs were ruling Egypt. ❌
During 3000 BC, pharaohs were ruling Egypt. ✅ 
Pharaohs have ruled Egypt. ❌